### ISSUE TITLE
Priorities of arrow operators variants

### ISSUE THREAD

**Issue by eshu** [scala/bug#11210]
There are different priorities for two variants of arrow operators:
```
$ scala
Welcome to Scala 2.12.7 (OpenJDK 64-Bit Server VM, Java 1.8.0_181).
Type in expressions for evaluation. Or try :help.

scala> "a" -> "b" -> "c"
res0: ((String, String), String) = ((a,b),c)

scala> "a" → "b" → "c"
res1: ((String, String), String) = ((a,b),c)

scala> "a" → "b" -> "c"
res2: ((String, String), String) = ((a,b),c)

scala> "a" -> "b" → "c"
res3: (String, (String, String)) = (a,(b,c))

scala> "a" + "b" -> "c"
res4: (String, String) = (ab,c)

scala> "a" + "b" → "c"
res5: String = a(b,c)
```
Unicode arrow "→" has higher priority then "->". Moreover there are some operators that have priority higher than "->" and lower than "→". It was unexpected for me when I got a broken code after replacement "->" with "→".

**Comment by eed3si9n** [comment]
[SLS 6.12.3](https://www.scala-lang.org/files/archive/spec/2.12/06-expressions.html#infix-operations):

> The _precedence_ of an infix operator is determined by the operator's first character. Characters are listed below in increasing order of precedence, with characters on the same line having the same precedence.

According to the list `(all other special characters)` has the highest precedence, then `* / %`, then `+ -`, and so on.

I don't have any suggestion on how this _should_ be fixed, but the current behavior seems consistent with the language spec to me.

**Comment by eshu** [comment]
Formally speaking it is true, but it breaks principle of least astonishment. I think replacement operators with their synonyms should not break code.

**Comment by eed3si9n** [comment]
All this is astonishing, worthy of a Scala Puzzler. But I can't imagine the degree of surprise it would cause if some existing code was relying on the behavior, and it changed between Scala 2 and 3.

**Comment by smarter** [comment]
Can we just deprecate the fancy unicode operators? They've never really pulled their weight, but they seem especially useless now that coding fonts with ligatures are a thing (e.g. https://github.com/tonsky/FiraCode).

**Comment by SethTisue** [comment]
> Can we just deprecate the fancy unicode operators?

+1

**Comment by som-snytt** [comment]
Duplicates https://github.com/scala/bug/issues/7613 which popped up in my beta bug search box on github, designed to avoid duplicates.

**Comment by SethTisue** [comment]
ah, good catch. let's keep this one; I closed #7613
